Fixes #887 - Add preview flag to python

ankitkumarr opened the pull request:
Adding `(preview)` while displaying the selection wizard on `func init` without affecting the parameter `--worker-runtime`. `func init --worker-runtime python` should still be valid.

fabiocav commented on src/Azure.Functions.Cli/Helpers/WorkerRuntimeLanguageHelper.cs:
```diff
@@ -42,6 +42,17 @@ public static class WorkerRuntimeLanguageHelper
             { WorkerRuntime.powershell, "powershell" }
         };
 
+        public static IDictionary<WorkerRuntime, string> WorkerToDisplayStrings => availableWorkersRuntime.Keys.ToDictionary(wr => wr, wr =>
```
nit: because of the lambdas, I think the expression bodied member here is impacting readability a bit, can we just switch and have a body instead?

ankitkumarr replied:
Sounds good, will update

fabiocav commented on src/Azure.Functions.Cli/Helpers/WorkerRuntimeLanguageHelper.cs:
```diff
@@ -42,6 +42,17 @@ public static class WorkerRuntimeLanguageHelper
+            switch (wr)
+            {
+                case WorkerRuntime.python:
+                    return "python (preview)";
```
Wondering if we should have a dictionary (WorkerRuntime/string) with runtime and display strings so we could properly format them (including proper casing and runtime name, like `.NET` instead of `dotnet`)

ankitkumarr replied:
That is a good call. I wanted to be least invasive and did not want to add another dictionary to maintain if a new WorkerRuntime is added or changed.
But, I think it should be ok as it's all in the same place and if you think we should change the display strings.

fabiocav replied:
Yeah, I get where you were coming from, just wanted to share that thought as I think it will help us provide a better experience than just the enum `ToString()` result

ankitkumarr replied:
The only other thing I could think of is we will either have to
1) have an inconsistency between `--worker-runtime` and the display strings. (`--worker-runtime dotnet` vs display string being `.NET`)
2) make changes to ensure params `--worker-runtime .NET` work.

fabiocav replied:
That's a good point. I was looking at this as just plain human readable runtime names, but I guess people could use what's displayed for the parameter/setting value as well. That's actually a good argument for leaving this as is.

ankitkumarr replied:
Yeah, that makes sense. Sounds good. I could open a separate PR that changes the display strings and accepts both, the display string and what we have now to be passed as `--worker-runtime`

fabiocav replied:
Sounds good

ankitkumarr commented:
@fabiocav, I addressed your comment yesterday and force pushed. After removing the lambda, it didn't make sense to keep it a Getter property so I just made it to a function. 
Let me know if I misunderstood or didn't address it properly.

fabiocav commented:
Looks good. Feel free to merge
